src/components/Exercise/index.tsx

```diff
@@ -242,6 +280,18 @@ export const Exercise = styled(({
                   props.canUpdateCurrentStep : !(i + 1 === exercise.questions.length)
               }
               previewMode={previewMode}
+              rightSideSlot={
+                React.isValidElement(rightSideSlot)
+                  ? React.cloneElement(
+                      rightSideSlot,
```

Comment:
Would the cloning become a problem if this element is re-rendered? Maybe the old state would be lost? I'm not 100% sure though.